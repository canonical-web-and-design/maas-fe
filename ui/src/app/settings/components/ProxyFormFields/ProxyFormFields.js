import PropTypes from "prop-types";
import React from "react";

import Button from "app/base/components/Button";
import Form from "app/base/components/Form";
import FormikField from "app/base/components/FormikField";

const ProxyFormFields = ({ formikProps }) => {
  const { handleSubmit, isSubmitting, touched, values } = formikProps;
  const disabled = isSubmitting || !Object.keys(touched).length;

  return (
    <Form onSubmit={handleSubmit}>
      <FormikField
        formikProps={formikProps}
        fieldKey="proxyType"
        value="noProxy"
        label="Don't use a proxy"
        type="radio"
        checked={values.proxyType === "noProxy"}
      />
      <FormikField
        formikProps={formikProps}
        fieldKey="proxyType"
        value="builtInProxy"
        label="MAAS built-in"
        type="radio"
        checked={values.proxyType === "builtInProxy"}
      />
      <FormikField
        formikProps={formikProps}
        fieldKey="proxyType"
        value="externalProxy"
        label="External"
        type="radio"
        checked={values.proxyType === "externalProxy"}
      />
      {values.proxyType === "externalProxy" && (
        <FormikField
          formikProps={formikProps}
          fieldKey="httpProxy"
          help="Enter the external proxy URL MAAS will use to download images and machines to download APT packages."
          type="text"
          required={true}
        />
      )}
      <FormikField
        formikProps={formikProps}
        fieldKey="proxyType"
        value="peerProxy"
        label="Peer"
        type="radio"
        checked={values.proxyType === "peerProxy"}
      />
      {values.proxyType === "peerProxy" && (
        <FormikField
          formikProps={formikProps}
          fieldKey="httpProxy"
          help="Enter the external proxy URL that the MAAS built-in proxy will use as an upstream cache peer. Machines will be configured to use MAAS' built-in proxy to download APT packages."
          type="text"
          required={true}
        />
      )}
      <div className="user-form__buttons">
        <Button
          appearance="positive"
          className="u-no-margin--bottom"
          type="submit"
          disabled={disabled}
        >
          Save
        </Button>
      </div>
    </Form>
  );
};

ProxyFormFields.propTypes = {
  formikProps: PropTypes.shape({
    errors: PropTypes.shape({
      httpProxy: PropTypes.string,
      proxyType: PropTypes.string
    }).isRequired,
    handleBlur: PropTypes.func.isRequired,
    handleChange: PropTypes.func.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    isSubmitting: PropTypes.bool,
    touched: PropTypes.shape({
      httpProxy: PropTypes.bool,
      proxyType: PropTypes.bool
    }).isRequired,
    values: PropTypes.shape({
      httpProxy: PropTypes.string,
      proxyType: PropTypes.string
    }).isRequired
  })
};

export default ProxyFormFields;
