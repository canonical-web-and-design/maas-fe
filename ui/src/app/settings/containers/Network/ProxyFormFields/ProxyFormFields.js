import PropTypes from "prop-types";
import React from "react";

import { extendFormikShape } from "app/settings/proptypes";
import FormikField from "app/base/components/FormikField";

const ProxyFormFields = ({ formikProps }) => {
  const { values } = formikProps;

  return (
    <>
      <label>
        HTTP proxy used by MAAS to download images, and by provisioned machines
        for APT and YUM packages.
      </label>
      <FormikField
        formikProps={formikProps}
        fieldKey="proxyType"
        value="noProxy"
        label="Don't use a proxy"
        type="radio"
        checked={values.proxyType === "noProxy"}
      />
      <FormikField
        formikProps={formikProps}
        fieldKey="proxyType"
        value="builtInProxy"
        label="MAAS built-in"
        type="radio"
        checked={values.proxyType === "builtInProxy"}
      />
      <FormikField
        formikProps={formikProps}
        fieldKey="proxyType"
        value="externalProxy"
        label="External"
        type="radio"
        checked={values.proxyType === "externalProxy"}
      />
      {values.proxyType === "externalProxy" && (
        <FormikField
          formikProps={formikProps}
          fieldKey="httpProxy"
          help="Enter the external proxy URL MAAS will use to download images and machines to download APT packages."
          type="text"
          required={true}
        />
      )}
      <FormikField
        formikProps={formikProps}
        fieldKey="proxyType"
        value="peerProxy"
        label="Peer"
        type="radio"
        checked={values.proxyType === "peerProxy"}
      />
      {values.proxyType === "peerProxy" && (
        <FormikField
          formikProps={formikProps}
          fieldKey="httpProxy"
          help="Enter the external proxy URL that the MAAS built-in proxy will use as an upstream cache peer. Machines will be configured to use MAAS' built-in proxy to download APT packages."
          type="text"
          required={true}
        />
      )}
    </>
  );
};

ProxyFormFields.propTypes = extendFormikShape({
  httpProxy: PropTypes.string,
  proxyType: PropTypes.string
});

export default ProxyFormFields;
