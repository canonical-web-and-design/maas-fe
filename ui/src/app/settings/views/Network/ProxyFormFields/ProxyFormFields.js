import { useFormikContext } from "formik";
import React from "react";

import FormikField from "app/base/components/FormikField";

const ProxyFormFields = () => {
  const { values } = useFormikContext();

  return (
    <>
      <label>
        HTTP proxy used by MAAS to download images, and by provisioned machines
        for APT and YUM packages.
      </label>
      <FormikField
        name="proxyType"
        value="noProxy"
        label="Don't use a proxy"
        type="radio"
      />
      <FormikField
        name="proxyType"
        value="builtInProxy"
        label="MAAS built-in"
        type="radio"
      />
      <FormikField
        name="proxyType"
        value="externalProxy"
        label="External"
        type="radio"
      />
      {values.proxyType === "externalProxy" && (
        <FormikField
          name="httpProxy"
          help="Enter the external proxy URL MAAS will use to download images and machines to download APT packages."
          type="text"
          required={true}
        />
      )}
      <FormikField
        name="proxyType"
        value="peerProxy"
        label="Peer"
        type="radio"
      />
      {values.proxyType === "peerProxy" && (
        <FormikField
          name="httpProxy"
          help="Enter the external proxy URL that the MAAS built-in proxy will use as an upstream cache peer. Machines will be configured to use MAAS' built-in proxy to download APT packages."
          type="text"
          required={true}
        />
      )}
    </>
  );
};

export default ProxyFormFields;
