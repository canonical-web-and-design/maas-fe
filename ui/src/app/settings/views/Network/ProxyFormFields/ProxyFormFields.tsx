import { useFormikContext } from "formik";

import type { ProxyFormValues } from "../ProxyForm/types";

import FormikField from "app/base/components/FormikField";

const ProxyFormFields = (): JSX.Element => {
  const { values } = useFormikContext<ProxyFormValues>();

  return (
    <>
      <p>
        HTTP proxy used by MAAS to download images, and by provisioned machines
        for APT and YUM packages.
      </p>
      <FormikField
        name="proxyType"
        value="noProxy"
        label="Don't use a proxy"
        type="radio"
      />
      <FormikField
        name="proxyType"
        value="builtInProxy"
        label="MAAS built-in"
        type="radio"
      />
      <FormikField
        name="proxyType"
        value="externalProxy"
        label="External"
        type="radio"
      />
      {values.proxyType === "externalProxy" && (
        <FormikField
          name="httpProxy"
          help="Enter the external proxy URL MAAS will use to download images and machines to download APT packages."
          type="text"
          required={true}
        />
      )}
      <FormikField
        name="proxyType"
        value="peerProxy"
        label="Peer"
        type="radio"
      />
      {values.proxyType === "peerProxy" && (
        <FormikField
          name="httpProxy"
          help="Enter the external proxy URL that the MAAS built-in proxy will use as an upstream cache peer. Machines will be configured to use MAAS' built-in proxy to download APT packages."
          type="text"
          required={true}
        />
      )}
    </>
  );
};

export default ProxyFormFields;
