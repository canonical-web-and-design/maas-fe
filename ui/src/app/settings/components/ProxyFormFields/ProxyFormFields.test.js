import { shallow } from "enzyme";
import React from "react";

import ProxyFormFields from "./ProxyFormFields";

describe("ProxyFormFields", () => {
  let formikProps;

  beforeEach(() => {
    formikProps = {
      errors: {},
      handleBlur: jest.fn(),
      handleChange: jest.fn(),
      handleSubmit: jest.fn(),
      touched: {},
      values: {}
    };
  });

  it("can render", () => {
    const wrapper = shallow(<ProxyFormFields formikProps={formikProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("disables the submit button when submitting", () => {
    formikProps.isSubmitting = true;
    const wrapper = shallow(<ProxyFormFields formikProps={formikProps} />);
    expect(
      wrapper
        .find("Button")
        .at(0)
        .prop("disabled")
    ).toBe(true);
  });
});
