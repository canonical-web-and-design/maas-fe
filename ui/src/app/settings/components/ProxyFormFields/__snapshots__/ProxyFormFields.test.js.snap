// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProxyFormFields can render 1`] = `
<Form
  onSubmit={[MockFunction]}
>
  <FormikField
    checked={false}
    fieldKey="proxyType"
    formikProps={
      Object {
        "errors": Object {},
        "handleBlur": [MockFunction],
        "handleChange": [MockFunction],
        "handleSubmit": [MockFunction],
        "touched": Object {},
        "values": Object {},
      }
    }
    label="Don't use a proxy"
    type="radio"
    value="noProxy"
  />
  <FormikField
    checked={false}
    fieldKey="proxyType"
    formikProps={
      Object {
        "errors": Object {},
        "handleBlur": [MockFunction],
        "handleChange": [MockFunction],
        "handleSubmit": [MockFunction],
        "touched": Object {},
        "values": Object {},
      }
    }
    label="MAAS built-in"
    type="radio"
    value="builtInProxy"
  />
  <FormikField
    checked={false}
    fieldKey="proxyType"
    formikProps={
      Object {
        "errors": Object {},
        "handleBlur": [MockFunction],
        "handleChange": [MockFunction],
        "handleSubmit": [MockFunction],
        "touched": Object {},
        "values": Object {},
      }
    }
    label="External"
    type="radio"
    value="externalProxy"
  />
  <FormikField
    checked={false}
    fieldKey="proxyType"
    formikProps={
      Object {
        "errors": Object {},
        "handleBlur": [MockFunction],
        "handleChange": [MockFunction],
        "handleSubmit": [MockFunction],
        "touched": Object {},
        "values": Object {},
      }
    }
    label="Peer"
    type="radio"
    value="peerProxy"
  />
  <div
    className="user-form__buttons"
  >
    <Button
      appearance="positive"
      className="u-no-margin--bottom"
      disabled={true}
      type="submit"
    >
      Save
    </Button>
  </div>
</Form>
`;
